ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,339 @@
+* [Secure backends](#secure-backends)
+* [Whitelist source range](#whitelist-source-range)
+* [Allowed parameters in configuration config map](#allowed-parameters-in-configuration-configmap)
+* [Default configuration options](#default-configuration-options)
+* [Websockets](#websockets)
+* [Optimizing TLS Time To First Byte (TTTFB)](#optimizing-tls-time-to-first-byte-tttfb)
+* [Retries in no idempotent methods](#retries-in-no-idempotent-methods)
+
+
+### Customizing nginx
+
+there are 3 ways to customize nginx
+
+1. config map: create a stand alone config map, use this if you want a different global configuration
+2. annoations: [annotate the ingress](#annotations), use this if you want a specific configuration for the site defined in the ingress rule
+3. custom template: when do people want this?
+
+
+#### Custom NGINX configuration
+
+Using a config map it is possible to customize the defaults in nginx.
+
+Please check the [custom configuration](examples/custom-configuration/README.md) example
+
+#### Annotations
+
+The following annotaitons are supported:
+
+|Name                 |type|
+|---------------------------|------|
+|[ingress.kubernetes.io/rewrite-target](#rewrite)|URI|
+|[ingress.kubernetes.io/add-base-url](#rewrite)|true or false|
+|[ingress.kubernetes.io/limit-connections](#rate-limiting)|number|
+|[ingress.kubernetes.io/limit-rps](#rate-limiting)|number|
+|[ingress.kubernetes.io/auth-type](#authentication)|basic or digest|
+|[ingress.kubernetes.io/auth-secret](#authentication)|string|
+|[ingress.kubernetes.io/auth-realm](#authentication)|string|
+|[ingress.kubernetes.io/ssl-redirect](#server-side-https-enforcement-through-redirect)|true or false|
+|[ingress.kubernetes.io/upstream-max-fails](#custom-nginx-upstream-checks)|number|
+|[ingress.kubernetes.io/upstream-fail-timeout](#custom-nginx-upstream-checks)|number|
+|[ingress.kubernetes.io/secure-backends](#secure-backends)|true or false|
+|[ingress.kubernetes.io/whitelist-source-range](#whitelist-source-range)|CIDR|
+
+
+
+#### Custom NGINX template
+
+The NGINX template is located in the file `/etc/nginx/template/nginx.tmpl`. Mounting a volume is possible to use a custom version.
+Use the [custom-template](examples/custom-template/README.md) example as a guide
+
+**Please note the template is tied to the go code. Be sure to no change names in the variable `$cfg`**
+
+
+### Custom NGINX upstream checks
+
+NGINX exposes some flags in the [upstream configuration](http://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream) that enables the configuration of each server in the upstream. The ingress controller allows custom `max_fails` and `fail_timeout` parameters in a global context using `upstream-max-fails` or `upstream-fail-timeout` in the NGINX config map or in a particular Ingress rule. By default this values are 0. This means NGINX will respect the `readinessProbe`, if is defined. If there is no probe, NGINX will not mark a server inside an upstream down.
```

Comment:
please add a comma after config map before or
s/by default this values are 0/it defautls to 0/